packages/iov-core/src/jsonrpcsigningserver.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { SigningServerCore } from "./signingservercore";
+
+const { fromHex } = Encoding;
+
+function pendingWithoutBnsd(): void {
+  if (!process.env.BNSD_ENABLED) {
+    pending("Set BNSD_ENABLED to enable bnsd-based tests");
+  }
+}
+
+function pendingWithoutEthereum(): void {
+  if (!process.env.ETHEREUM_ENABLED) {
+    pending("Set ETHEREUM_ENABLED to enable ethereum-based tests");
+  }
+}
+
+async function randomBnsAddress(): Promise<Address> {
+  const rawKeypair = await Ed25519.makeKeypair(await Random.getBytes(32));
+  const randomIdentity: PublicIdentity = {
+    chainId: "some-testnet" as ChainId,
+    pubkey: {
+      algo: Algorithm.Ed25519,
+      data: rawKeypair.pubkey as PublicKeyBytes,
+    },
+  };
+  return bnsCodec.identityToAddress(randomIdentity);
+}
+
+const bnsdUrl = "ws://localhost:22345";
+const bnsdFaucetMnemonic = "degree tackle suggest window test behind mesh extra cover prepare oak script";
+const ethereumChainId = "5777" as ChainId;
+const ganacheMnemonic = "oxygen fall sure lava energy veteran enroll frown question detail include maximum";
+
+async function makeJsonRpcSigningServer(): Promise<JsonRpcSigningServer> {
+  const profile = new UserProfile();
+  const ed25519Wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(bnsdFaucetMnemonic));
+  const secp256k1Wallet = profile.addWallet(Secp256k1HdWallet.fromMnemonic(ganacheMnemonic));
+  const signer = new MultiChainSigner(profile);
+
+  const bnsConnection = (await signer.addChain(bnsConnector(bnsdUrl))).connection;
+
+  // faucet identity
+  await profile.createIdentity(ed25519Wallet.id, bnsConnection.chainId(), HdPaths.simpleAddress(0));
+  // ganache second identity
+  await profile.createIdentity(secp256k1Wallet.id, ethereumChainId, HdPaths.bip44(60, 0, 0, 1));
+
+  return new JsonRpcSigningServer(new SigningServerCore(profile, signer));
+}
+
+describe("JsonRpcSigningServer", () => {
+  const ganacheSecondIdentity: PublicIdentity = {
+    chainId: ethereumChainId,
+    pubkey: {
+      algo: Algorithm.Secp256k1,
+      data: Encoding.fromHex(
+        "041d4c015b00cbd914e280b871d3c6ae2a047ca650d3ecea4b5246bb3036d4d74960b7feb09068164d2b82f1c7df9e95839b29ae38e90d60578b2318a54e108cf8",
+      ) as PublicKeyBytes,
+    },
+  };
+
+  const defaultAmount: Amount = {
+    quantity: "1",
+    fractionalDigits: 9,
+    tokenTicker: "CASH" as TokenTicker,
+  };
+
+  it("can get bnsd identities", async () => {
+    pendingWithoutBnsd();
+
+    const bnsConnection = await bnsConnector(bnsdUrl).client();
+
+    const server = await makeJsonRpcSigningServer();
+
+    const response = await server.handleUnchecked({
+      jsonrpc: "2.0",
+      id: 123,
+      method: "getIdentities",
+      params: {
+        reason: "Who are you?",
+        chainIds: [bnsConnection.chainId()],
+      },
+    });
+    expect(response.jsonrpc).toEqual("2.0");
+    expect(response.id).toEqual(123);
+    if (isJsonRpcErrorResponse(response)) {
```

Comment:
style q:

why not use `expect(isJsonRpcErrorResponse(response)).toEqual(true)` here?

Reply:
response is of type `JsonRpcResponse | JsonRpcErrorResponse`. I need to throw here to allow accessing members of `JsonRpcResponse` that do not exist in `JsonRpcErrorResponse`